src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +258,107 @@ pub fn register(registry: &mut FunctionRegistry) {
                     .any(|(k, _)| k == key)
             },
         );
+
+    registry.register_function_factory("map_pick", |_, args_type: &[DataType]| {
+        if args_type.len() < 2 {
+            return None;
+        }
+
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "map_pick".to_string(),
+                args_type: args_type.to_vec(),
+                return_type: args_type[0].clone(),
+            },
+            eval: FunctionEval::Scalar {
+                calc_domain: Box::new(move |_, _| FunctionDomain::Full),
+                eval: Box::new(map_pick_fn_vec),
+            },
+        }))
+    });
+
+    fn map_pick_fn_vec(args: &[ValueRef<AnyType>], _: &mut EvalContext) -> Value<AnyType> {
+        let len = args.iter().find_map(|arg| match arg {
+            ValueRef::Column(col) => Some(col.len()),
+            _ => None,
+        });
+
+        let source_data_type = match args.first().unwrap() {
+            ValueRef::Scalar(s) => s.infer_data_type(),
+            ValueRef::Column(c) => c.data_type(),
+        };
+
+        let source_map = match &args[0] {
+            ValueRef::Scalar(ScalarRef::Map(s)) => {
+                KvPair::<GenericType<0>, GenericType<1>>::try_downcast_column(s).unwrap()
+            }
+            ValueRef::Column(Column::Map(c)) => {
+                KvPair::<GenericType<0>, GenericType<1>>::try_downcast_column(&c.values).unwrap()
+            }
+            _ => unreachable!(),
+        };
+
+        let mut builder: ArrayColumnBuilder<KvPair<GenericType<0>, GenericType<1>>> =
+            ArrayType::create_builder(
+                args.len() - 1,
+                &source_data_type.as_map().unwrap().as_tuple().unwrap(),
+            );
+        for key_arg in args[1..].iter() {
```

Comment:
If the argument is a column, it will have multiple rows, we need to iterate from 0 to `len`.